### DISCUSSION TITLE
Specific build image per OS e.g. Apple M1 vs Apple Intel

### DISCUSSION THREAD

**Discussion by ex-pexa-abruno** [batect/batect#1203] [Ideas]
Is there a way to assign a specific docker image for a build depending on OS?

```
containers:
  build-env:
    image: amazoncorretto:11-alpine
```

The team is using `amazoncorretto:11-alpine` and it has been approved by security, and production.

However it does not work on the Apple M1.  See https://github.com/corretto/corretto-docker/issues/46

So I was wondering if there is a way to assign a different image tied to the OS batect is running on?

P.S.  I have had success with `adoptopenjdk/openjdk11:jdk11u-nightly-slim`

**Comment by charleskorn** [comment]
Hey @pexa-abruno, at present, this isn't something supported by Batect. 

However, you could create your own multi-architecture image that uses Coretto on x86 and the OpenJDK image on ARM. The easiest way is with a manifest, which you can create with `docker manifest create`. 

There's an example of this in [this blog post](https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/) under "The hard way with docker manifest". Something like this _should_ work:

```
docker manifest create \
  pexa-abruno/jdk:11 \
  --amend amazoncorretto:11-alpine \
  --amend adoptopenjdk/openjdk11:aarch64-centos-jdk11u-nightly-slim
```
